### PR TITLE
Ignore darwin-arm64

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
Some plugins have begun building `darwin-arm64` in addition to `darwin-amd64`. Update the matterbuild bundling heuristic for pre-packaged plugins to ignore `darwin-arm64` when present for now, as we have no intention to start supporting Apple M1 for the overall server platform.

#### Ticket Link
N/A

### CODE REVIEW

File: server/plugin_release.go [hunk 1]
```diff
@@ -399,7 +399,7 @@ func createPlatformPlugins(repositoryName, tag, pluginFilePath, pluginFolder str
 	}
 
 	platformFileExclusion := map[string][]string{
-		"osx-amd64":     {"windows", "linux"},
+		"osx-amd64":     {"windows", "linux", "darwin-arm64"},
```
Comment (lieut-data) [inline server/plugin_release.go:402]: Not visible in the `mattermost-plugin-demo` below is the presence of a faked `darwin-arm64` binary that triggers the tests (and fails until this change is made).

**Review by cpanato — APPROVED** [review]

**Comment by cpanato** [comment]
when approved let me know, then i can merge and deploy that

**Review by agarciamontoro — APPROVED** [review]
@cpanato, all yours :)

**Comment by cpanato** [comment]
@lieut-data @agarciamontoro deployed

**Comment by lieut-data** [comment]
Thank you, @cpanato!

**Pull request merged**